### PR TITLE
Added feature PickerInputRow

### PR DESCRIPTION
**Pull request by SpencerC**
Followed pattern of the generic picker input

#808

### CODE REVIEW

**Comment by SpencerC** [comment]
Also added a variable to control the spacing between the keyboard and the selected field when scrolling is required, and added smoothly animated scrolling. Tried to make it another pull request, but github desktop is misbehaving. LMK if that's a problem, and I'll try again

File: Eureka.xcworkspace/xcshareddata/Eureka.xcscmblueprint [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{
```
Comment (mtnbarreto) [inline Eureka.xcworkspace/xcshareddata/Eureka.xcscmblueprint]: This file should be added to gitignore.

File: Source/Core/Core.swift [hunk 2]
```diff
@@ -840,7 +846,7 @@ extension FormViewController {
             table.contentInset = tableInsets
             table.scrollIndicatorInsets = scrollIndicatorInsets
             if let selectedRow = table.indexPath(for: cell) {
-                table.scrollToRow(at: selectedRow, at: .none, animated: false)
+                table.scrollToRow(at: selectedRow, at: .none, animated: animateScroll)
```
Comment (mtnbarreto) [inline Source/Core/Core.swift]: I'm not sure of passing true as animated value works in all cases. In particular when we navigate to a row that is not being displayed in the screen. Have you tested it?

File: Source/Core/Core.swift [hunk 3]
```diff
@@ -431,6 +431,12 @@ open class FormViewController : UIViewController, FormViewControllerProtocol {
         }
     }
     
+    /// Extra space to leave between between the row in focus and the keyboard
+    open var rowKeyboardSpacing: CGFloat = 0
```
Comment (mtnbarreto) [inline Source/Core/Core.swift]: 👍 Please add associated documentation in readme.

File: Source/Core/Core.swift (hunk 2 above)
Comment (mtnbarreto) [inline Source/Core/Core.swift]: i'm fine with this if it works. Please document it in readme.

File: Source/Rows/PickerInputRow.swift [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import Foundation
+
+class PickerInputTextField: UITextField {
```
Comment (mtnbarreto) [inline Source/Rows/PickerInputRow.swift]: We don't need to add a UITextField to present the picker control. Take a look at `DateCell` that also has a picker and shows it by implementing these 2 methods....

```swift
 open override func cellCanBecomeFirstResponder() -> Bool {
        return canBecomeFirstResponder
    }

    override open var canBecomeFirstResponder: Bool {
        get {
            return !row.isDisabled
        }
    }
```

By doing so we can use the standard UIKit table view cell.

File: Source/Rows/PickerInputRow.swift [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+        textField.clearButtonMode = .never
+        if row.isHighlighted {
+            textFieldStartColor = textField.textColor
+            textField.textColor = textField.tintColor
+        } else if textFieldStartColor != nil {
+            textField.textColor = textFieldStartColor
+        }
+        
+        textLabel?.text = nil
+        detailTextLabel?.text = nil
```
Comment (mtnbarreto) [inline Source/Rows/PickerInputRow.swift]: if row.title is set up we should show the value using `detailTextLabel` otherwise use the `textLabel`.

**Comment by SpencerC** [comment]
@mtnbarreto All should be updated per your feedback. Let me know how it looks!

**Comment by mtnbarreto** [comment]
Hi @SpencerC! Now it looks fine! Great work!!!

**Pull request merged**